### ISSUE TITLE
[question] Move jails with zfs send not working anymore?

### ISSUE THREAD

**Issue by kasper5** [iocage/iocage#51]
Hi,

Moving jails with zfs send doesn't seem to work anymore when a jail has been created with the new iocage.

Do the new iocage create clones as default instead of fully independent jails?

**Comment by skarekrow** [comment]
Yes clones are the default for all jails. Do you require independent jails? Clones shouldn't pose a problem here that I'm aware of.

**Comment by kasper5** [comment]
If I am to able move jails with zfs send/recv I don't really care what kind of jail type it is. If I try to move a jail with zfs send as it looks now I'll get an error message like "clone origin at destination not existing", I can't really remember the exact message. But if you want to see the message I'll give it to you.

Is there a way to get around this when working with clones?

BTW, great work on iocage I really like it.

**Comment by skarekrow** [comment]
Thanks for the kind words!

Sure next week I'll try to mock up an example to show you how to move the jails. ZFS should send the clone and it's origin when you send. I just can't remember the exact syntax offhand. I may just take this opportunity to port the send feature I developed with the older iocage but that could take longer then I have this week. 

But I'll try to touch base with you and give you the example before the weeks end :)

**Comment by kasper5** [comment]
That's awesome, thank you! :) ... and don't stress, take your time :)

**Comment by skarekrow** [comment]
@kasper5 Ok tested and it all works quite easy :)

```
NAME                                      USED  AVAIL  REFER  MOUNTPOINT
ioc-test                                  916K  48.2G    19K  /ioc-test
zroot                                    1.89G  54.1G    96K  /zroot
zroot/ROOT                               1.25G  54.1G    96K  none
zroot/ROOT/default                       1.25G  54.1G  1.25G  /
zroot/iocage                              468M  54.1G   976K  /iocage
zroot/iocage/download                     112M  54.1G    96K  /iocage/download
zroot/iocage/download/11.0-RELEASE        112M  54.1G   112M  /iocage/download/11.0-RELEASE
zroot/iocage/images                        96K  54.1G    96K  /iocage/images
zroot/iocage/jails                         96K  54.1G    96K  /iocage/jails
zroot/iocage/log                          100K  54.1G   100K  /iocage/log
zroot/iocage/releases                     355M  54.1G    96K  /iocage/releases
zroot/iocage/releases/11.0-RELEASE        355M  54.1G    96K  /iocage/releases/11.0-RELEASE
zroot/iocage/releases/11.0-RELEASE/root   355M  54.1G   355M  /iocage/releases/11.0-RELEASE/root
zroot/iocage/templates                     96K  54.1G    96K  /iocage/templates
zroot/tmp                                 120K  54.1G   120K  /tmp
zroot/usr                                 179M  54.1G    96K  /usr
zroot/usr/home                            178M  54.1G   178M  /usr/home
zroot/usr/ports                            96K  54.1G    96K  /usr/ports
zroot/usr/src                              96K  54.1G    96K  /usr/src
zroot/var                                 980K  54.1G    96K  /var
zroot/var/audit                            96K  54.1G    96K  /var/audit
zroot/var/crash                            96K  54.1G    96K  /var/crash
zroot/var/log                             476K  54.1G   476K  /var/log
zroot/var/mail                            120K  54.1G   120K  /var/mail
zroot/var/tmp                              96K  54.1G    96K  /var/tmp
root@iocage:~ # iocage create tag="MOVEME" -r 11.0-RELEASE
4ddfbe07-b163-40b0-a62d-c9c22ec2e80a (MOVEME) successfully created!
root@iocage:~ # iocage snapshot -n moving MOVEME
Snapshot: zroot/iocage/jails/4ddfbe07-b163-40b0-a62d-c9c22ec2e80a@moving created.
root@iocage:~ # zfs send -R zroot/iocage/releases/11.0-RELEASE@4ddfbe07-b163-40b0-a62d-c9c22ec2e80a | zfs recv -vd ioc-test
WARNING: could not send zroot/iocage/releases/11.0-RELEASE@4ddfbe07-b163-40b0-a62d-c9c22ec2e80a: does not exist
WARNING: could not send zroot/iocage/releases/11.0-RELEASE@4ddfbe07-b163-40b0-a62d-c9c22ec2e80a: does not exist
receiving full stream of zroot/iocage/releases/11.0-RELEASE/root@4ddfbe07-b163-40b0-a62d-c9c22ec2e80a into ioc-test/iocage/releases/11.0-RELEASE/root@4ddfbe07-b163-40b0-a62d-c9c22ec2e80a
received 620MB stream in 5 seconds (124MB/sec)
root@iocage:~ # zfs send -R zroot/iocage/jails/4ddfbe07-b163-40b0-a62d-c9c22ec2e80a@moving | zfs recv -vd ioc-test
receiving full stream of zroot/iocage/jails/4ddfbe07-b163-40b0-a62d-c9c22ec2e80a@moving into ioc-test/iocage/jails/4ddfbe07-b163-40b0-a62d-c9c22ec2e80a@moving
received 53.0KB stream in 1 seconds (53.0KB/sec)
found clone origin ioc-test/iocage/releases/11.0-RELEASE/root@4ddfbe07-b163-40b0-a62d-c9c22ec2e80a
receiving incremental stream of zroot/iocage/jails/4ddfbe07-b163-40b0-a62d-c9c22ec2e80a/root@moving into ioc-test/iocage/jails/4ddfbe07-b163-40b0-a62d-c9c22ec2e80a/root@moving
received 34.6KB stream in 1 seconds (34.6KB/sec)
NAME                                                              USED  AVAIL  REFER  MOUNTPOINT
ioc-test                                                          316M  47.9G    19K  /ioc-test
ioc-test/iocage                                                   316M  47.9G    19K  /ioc-test/iocage
ioc-test/iocage/jails                                            75.5K  47.9G    19K  /ioc-test/iocage/jails
ioc-test/iocage/jails/4ddfbe07-b163-40b0-a62d-c9c22ec2e80a       56.5K  47.9G    22K  /ioc-test/iocage/jails/4ddfbe07-b163-40b0-a62d-c9c22ec2e80a
ioc-test/iocage/jails/4ddfbe07-b163-40b0-a62d-c9c22ec2e80a/root  34.5K  47.9G   315M  /ioc-test/iocage/jails/4ddfbe07-b163-40b0-a62d-c9c22ec2e80a/root
ioc-test/iocage/releases                                          315M  47.9G    19K  /ioc-test/iocage/releases
ioc-test/iocage/releases/11.0-RELEASE                             315M  47.9G    19K  /ioc-test/iocage/releases/11.0-RELEASE
ioc-test/iocage/releases/11.0-RELEASE/root                        315M  47.9G   315M  /ioc-test/iocage/releases/11.0-RELEASE/root
zroot                                                            1.89G  54.1G    96K  /zroot
zroot/ROOT                                                       1.25G  54.1G    96K  none
zroot/ROOT/default                                               1.25G  54.1G  1.25G  /
zroot/iocage                                                      468M  54.1G   976K  /iocage
zroot/iocage/download                                             112M  54.1G    96K  /iocage/download
zroot/iocage/download/11.0-RELEASE                                112M  54.1G   112M  /iocage/download/11.0-RELEASE
zroot/iocage/images                                                96K  54.1G    96K  /iocage/images
zroot/iocage/jails                                                312K  54.1G    96K  /iocage/jails
zroot/iocage/jails/4ddfbe07-b163-40b0-a62d-c9c22ec2e80a           216K  54.1G   100K  /iocage/jails/4ddfbe07-b163-40b0-a62d-c9c22ec2e80a
zroot/iocage/jails/4ddfbe07-b163-40b0-a62d-c9c22ec2e80a/root      116K  54.1G   355M  /iocage/jails/4ddfbe07-b163-40b0-a62d-c9c22ec2e80a/root
zroot/iocage/log                                                  100K  54.1G   100K  /iocage/log
zroot/iocage/releases                                             355M  54.1G    96K  /iocage/releases
zroot/iocage/releases/11.0-RELEASE                                355M  54.1G    96K  /iocage/releases/11.0-RELEASE
zroot/iocage/releases/11.0-RELEASE/root                           355M  54.1G   355M  /iocage/releases/11.0-RELEASE/root
zroot/iocage/templates                                             96K  54.1G    96K  /iocage/templates
zroot/tmp                                                         120K  54.1G   120K  /tmp
zroot/usr                                                         179M  54.1G    96K  /usr
zroot/usr/home                                                    178M  54.1G   178M  /usr/home
zroot/usr/ports                                                    96K  54.1G    96K  /usr/ports
zroot/usr/src                                                      96K  54.1G    96K  /usr/src
zroot/var                                                         980K  54.1G    96K  /var
zroot/var/audit                                                    96K  54.1G    96K  /var/audit
zroot/var/crash                                                    96K  54.1G    96K  /var/crash
zroot/var/log                                                     476K  54.1G   476K  /var/log
zroot/var/mail                                                    120K  54.1G   120K  /var/mail
zroot/var/tmp                                                      96K  54.1G    96K  /var/tmp
root@iocage:~ # zfs list -t snap
NAME                                                                              USED  AVAIL  REFER  MOUNTPOINT
ioc-test/iocage/jails/4ddfbe07-b163-40b0-a62d-c9c22ec2e80a@moving                    0      -    22K  -
ioc-test/iocage/jails/4ddfbe07-b163-40b0-a62d-c9c22ec2e80a/root@moving               0      -   315M  -
ioc-test/iocage/releases/11.0-RELEASE/root@4ddfbe07-b163-40b0-a62d-c9c22ec2e80a      0      -   315M  -
zroot/iocage/jails/4ddfbe07-b163-40b0-a62d-c9c22ec2e80a@moving                       0      -   100K  -
zroot/iocage/jails/4ddfbe07-b163-40b0-a62d-c9c22ec2e80a/root@moving                  0      -   355M  -
zroot/iocage/releases/11.0-RELEASE/root@4ddfbe07-b163-40b0-a62d-c9c22ec2e80a         0      -   355M  -
```

TLDR:

1. `iocage snapshot -n NAME JAIL`
2. ` zfs send -R POOL/iocage/releases/RELEASE@JAILUUID | zfs recv -d NEWPOOL`
3. ` zfs send -R POOL/iocage/jails/JAILUUID@SNAPSHOT | zfs recv -d NEWPOOL`


Hopefully this helps!
